drivers/sensor/apds9960/apds9960.c

```diff
@@ -0,0 +1,584 @@
+
+	/* Read value from CONFIG2 register */
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONFIG2_REG, &val) < 0) {
+		SYS_LOG_DBG(" Failed");
+		return false;
+	}
+
+	/* Set bits in register to given value */
+	boost &= 0x03;
+	boost = boost << 4;
+	val &= 0xcf;
+	val |= boost;
+
+	/* Write register value back into CONFIG2 register */
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONFIG2_REG, val) < 0) {
+		SYS_LOG_DBG(" Failed");
+		return false;
+	}
+
+	return true;
+}
+
+int apds9960_proxy_setup(struct device *dev, int gain)
+{
+	struct apds9960_data *data = dev->driver_data;
+
+	/* Power ON */
+	if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, BIT(0), 1) < 0) {
+		SYS_LOG_DBG("Power on bit not set.");
+		return -EIO;
+	}
+
+	/* ADC value */
+	if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ATIME_REG, APDS9960_ATIME_WRTIE, 0xB6) < 0) {
+		SYS_LOG_DBG("ADC bits are not written");
+		return -EIO;
+	}
+
+	/* proxy Gain */
+	if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONTROL_REG, (BIT(2) | BIT(3)), (gain & 0x0C)) < 0) {
+		SYS_LOG_DBG("proxy Gain is not set");
+		return -EIO;
+	}
+
+	/* Enable proxy */
+	if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, (BIT(2) | BIT(3)), 0x05) < 0) {
+		SYS_LOG_DBG("Proxy is not enabled");
+		return -EIO;
+	}
+
+	if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, BIT(5), 0x00) < 0) {
+		SYS_LOG_DBG("Proxy interrupt is not enabled");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int apds9960_ambient_setup(struct device *dev, int gain)
+{
+	struct apds9960_data *data = dev->driver_data;
+
+	/* Power ON */
+	if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, BIT(0), 1) < 0) {
+		SYS_LOG_DBG("Power on bit not set.");
+		return -EIO;
+	}
+
+	/* ADC value */
+	if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ATIME_REG, APDS9960_ATIME_WRTIE, 0xB6) < 0) {
+		SYS_LOG_DBG("ADC bits are not written");
+		return -EIO;
+	}
+
+	/* ALE Gain */
+	if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONTROL_REG, (BIT(0) | BIT(1)), (gain & 0x03)) < 0) {
+		SYS_LOG_DBG("proxy Gain is not set");
+		return -EIO;
+	}
+
+	if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		0x80, BIT(4), 0x00) < 0) {
+		return -EIO;
+	}
+
+	/* Enable ALE */
+	if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, (BIT(0) | BIT(1)), 0x03) < 0) {
+		SYS_LOG_DBG("Proxy is not enabled");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int apds9960_sensor_setup(struct device *dev, int gain)
+{
+	struct apds9960_data *data = dev->driver_data;
+
+	u8_t chip_id;
+
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ID_REG, &chip_id) < 0) {
+		SYS_LOG_DBG("failed reading chip id");
+		return -EIO;
+	}
+
+	if (!(chip_id == APDS9960_ID_1) || (chip_id == APDS9960_ID_2)) {
```

Comment:
As you wrote it, ! applies just for ID_1. I think you meant to do it for both of the IDs.